collector/config.go

```diff
@@ -62,6 +62,39 @@ type Prometheus struct {
 	MetricsConfig []string `json:"metrics_config"`
 }
 
+type Jolokia struct {
+	//the endpoint to hit to scrape metrics. Eg https://${HOST}/jolokia
+	Endpoint EndpointConfig `json:"endpoint"`
+
+	//holds names of different metrics that can be collected
+	MetricsConfig []JolokiaMetricConfig `json:"metrics_config"`
+
+	//the frequency at which metrics should be collected
+	PollingFrequency string `json:"polling_frequency"`
+}
+
+type JolokiaMetricConfig struct {
```

Comment:
I think there should be a common `MetricConfig` that has the fields shared by the prometheus `MetricConfig` and `JolokiaMetricConfig`, then create a `PrometheusMetricConfig`, and embed the common `MetricConfig` in both.